line-bot-spring-boot/src/main/java/com/linecorp/bot/spring/boot/support/LineMessageHandlerSupport.java

```diff
@@ -0,0 +1,218 @@
+    void refresh() {
+        final Map<String, Object> handlerBeanMap =
+                applicationContext.getBeansWithAnnotation(LineMessageHandler.class);
+
+        final List<HandlerMethod> collect = handlerBeanMap
+                .values().stream()
+                .flatMap((Object consumer) -> {
+                    final Method[] uniqueDeclaredMethods =
+                            ReflectionUtils.getUniqueDeclaredMethods(consumer.getClass());
+
+                    return stream(uniqueDeclaredMethods)
+                            .map((Method method) -> {
+                                final EventMapping mapping = getMergedAnnotation(method, EventMapping.class);
+                                if (mapping == null) {
+                                    return null;
+                                }
+
+                                Preconditions.checkState(method.getParameterCount() == 1,
+                                                         "Number of parameter should be 1. But {}",
+                                                         method.getParameterTypes());
+                                // TODO: Support more than 1 argument. Like MVC's argument resolver?
+
+                                final Type type = method.getGenericParameterTypes()[0];
+
+                                final Predicate<Event> predicate = new EventPredicate(type);
+                                return new HandlerMethod(predicate, consumer, method,
+                                                         getPriority(mapping, type));
+                            })
+                            .filter(Objects::nonNull);
+                })
+                .sorted(Ordering.natural().onResultOf(HandlerMethod::getPriority).reverse())
+                .collect(toList());
+
+        collect.forEach(item -> log.info("{} > {}", item.getSupportType(),
+                                         item.getHandler().toGenericString()));
+
+        eventConsumerList = collect;
+    }
+
+    private int getPriority(final EventMapping mapping, final Type type) {
+        if (mapping.priority() != EventMapping.DEFAULT_PRIORITY_VALUE) {
+            return mapping.priority();
+        }
+
+        if (type == Event.class) {
+            return EventMapping.DEFAULT_PRIORITY_FOR_EVENT_IFACE;
+        }
+
+        if (type instanceof Class) {
+            return ((Class) type).isInterface()
+                   ? EventMapping.DEFAULT_PRIORITY_FOR_IFACE
+                   : EventMapping.DEFAULT_PRIORITY_FOR_CLASS;
+        }
+
+        if (type instanceof ParameterizedType) {
+            return EventMapping.DEFAULT_PRIORITY_FOR_PARAMETRIZED_TYPE;
+        }
+
+        throw new IllegalStateException();
+    }
+
+    @Value
+    public static class HandlerMethod {
+        Predicate<Event> supportType;
+        Object object;
+        Method handler;
+        int priority;
+    }
+
+    @PostMapping("${line.bot.handler.path:/callback}")
```

Comment:
Okay, I see.
(we were discussed this issue a few days ago, but I forgot to submit the review comment :()